### PR TITLE
Add support for `getAsBinary` in `downloadObject()`

### PR DESCRIPTION
**Pull request by mbandizzle**
This is needed for downloading certain files like Microsoft Excel xlsx files.  Credit goes to Christopher Watson.

### CODE REVIEW

**Comment by lmajano** [comment]
Seems we have several issues on the tests. Especially on Adobe: 

```
-> Template:Attribute validation error for CFHTTP.:The value of the GETASBINARY attribute is invalid. The value must be one of YES, NO, NEVER, or AUTO.
```

**Comment by lmajano** [comment]
The entire build process fails at this point https://travis-ci.com/github/coldbox-modules/s3sdk/builds/231448516

File: models/AmazonS3.cfc [hunk 1]
```diff
@@ -888,7 +888,8 @@ component accessors="true" singleton {
 	struct function downloadObject(
 		required string bucketName = variables.defaultBucketName,
 		required string uri,
-		required string filepath
+		required string filepath,
+		boolean getAsBinary = false
```
Comment (lmajano) [inline models/AmazonS3.cfc]: Can you change to using `yes` and `no` due to Adobe being dumb on booleans please.

Comment (mbandizzle) [reply models/AmazonS3.cfc]: All set

**Pull request merged**